### PR TITLE
Fix GetRand due to that InitEntropy (srand) is not called.

### PR DESCRIPTION
**Pull request by kghost**
#### Problem
InitEntropy (srand) is not called, so GetRand in `src/lib/support/RandUtils.h` always return same value.

#### Change overview
* Implement InitEntropy and call it at start inside `PlatformManagerImpl::_InitChipStack`
* Remove legacy code which is not being used at all
* Reorder `chip::Crypto::add_entropy_source` before `PlatformManagerImpl::_InitChipStack` to ensure the DRBG is initialized before being used.

#### Testing
Manually verified using unit-tests.

### CODE REVIEW

File: src/platform/Entropy.cpp [hunk 1]
```diff
@@ -22,34 +22,18 @@
...
     unsigned int seed;
-
-    // Initialize the source used by CHIP to get secure random data.
-    err = Platform::Security::InitSecureRandomDataSource(getentropy, 64, NULL, 0);
-    SuccessOrExit(err);
-
-    // Seed the standard rand() pseudo-random generator with data from the secure random source.
-    err = Platform::Security::GetSecureRandomData((uint8_t *) &seed, sizeof(seed));
-    SuccessOrExit(err);
+    ReturnErrorOnFailure(chip::Crypto::DRBG_get_bytes((uint8_t *) &seed, sizeof(seed)));
```
Comment (andy31415) [inline src/platform/Entropy.cpp:34]: Previous code had a InitSecureRandomDataSource call which had a platform-dependent argument. Where is that gone?

Generally I expect the change description to explain why this PR removed files and methods and how the new code functions over the old.

**Review by andy31415 — CHANGES_REQUESTED** [review]
Requesting changes:

PR does more than just call 'init random'. It removes platform-specific entropy.cpp (which I am guessing they are never used?) and creates a generic platform one.

I expect this kind of change to be explained in the summary of the change: please explain what you do and how.

File: src/platform/Entropy.cpp (hunk 1 above)
Comment (kghost) [reply src/platform/Entropy.cpp:34]: `InitSecureRandomDataSource` has already been removed long ago. Now we got a platform independent `DRBG_get_bytes`

**Review by andy31415 — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Review by emargolis — APPROVED** [review]

**Comment by andy31415** [comment]
@bzbarsky-apple @chrisdecenzo @Damian-Nordic @hawk248 @jepenven-silabs ?

**Review by Damian-Nordic — APPROVED** [review]

**Pull request merged**